webview/src/App.tsx

```diff
@@ -64,22 +67,85 @@ function useTheme(initialThemeConfig: string) {
   return { theme, setThemeConfig };
 }
 
+const textEncoder = new TextEncoder();
+
+function onChange(initialSceneVersion: number, contentType: string) {
+  let previousSceneVersion = initialSceneVersion;
```

Comment:
this isn't really `previousSceneVersion` as it will always point to the `initialSceneVersion` (which the `onChange` function will always be called with). Is that intentional?

Reply:
I might be wrong, but here is the logic i went with:

- The `onChange` function is called inside a `useRef`, so it will only be called one time.
- it returns a closure, with the `previousSceneVersion` variable set to the `initialSceneVersion`
- each time the closure is called, the `previousSceneVersion` increase

Reply:
I agree this is not really readable, any suggestions are welcome ! I mostly struggle on avoiding recalling the `debounce` function on every render.

Reply:
I simplified the logic by passing the onChange function directly as a prop 35434d7